src/Azure.Functions.Cli/Common/AuthManager.cs

```diff
@@ -0,0 +1,312 @@
+            var middlewareAuthSettingsFile = SecretsManager.MiddlewareAuthSettingsFileName;
+            MiddlewareAuthSettings = new AuthSettingsFile(middlewareAuthSettingsFile);
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_AUTO_AAD", "True");
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_CLIENT_ID", clientId);
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_CLIENT_SECRET", clientSecret);
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_DEFAULT_PROVIDER", "AzureActiveDirectory");
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_ENABLED", "True");
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_OPENID_ISSUER", "https://sts.windows.net/" + tenant + "/");
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_RUNTIME_VERSION", "1.0.0");
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_TOKEN_STORE", "True");
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_UNAUTHENTICATED_ACTION", "AllowAnonymous");
+
+            // Middleware requires signing and encryption keys for local testing
+            // These will be different than the encryption and signing keys used by the application in production
+            string encryptionKey = ComputeSha256Hash(clientSecret);
+            string signingKey = ComputeSha256Hash(clientId);
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_ENCRYPTION_KEY", encryptionKey);
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_SIGNING_KEY", signingKey);
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_ALLOWED_AUDIENCES", serializedReplyUrls);
+            MiddlewareAuthSettings.Commit();
+        }
+
+        public Dictionary<string, string> CreateAuthSettingsToPublish(string homepage, string clientId, string clientSecret, string tenant, List<string> replyUrls)
+        {
+            // The 'allowedAudiences' setting of /config/authsettings is of the form ["{replyURL1}", "{replyURL2}"]
+            string serializedArray = JsonConvert.SerializeObject(replyUrls, Formatting.Indented);
+
+            var authSettingsToPublish = new Dictionary<string, string>();
+            authSettingsToPublish.Add("allowedAudiences", serializedArray);
+            authSettingsToPublish.Add("isAadAutoProvisioned", "True");
+            authSettingsToPublish.Add("clientId", clientId);
+            authSettingsToPublish.Add("clientSecret", clientSecret);
+            authSettingsToPublish.Add("defaultProvider", "0"); // 0 corresponds to AzureActiveDirectory
+            authSettingsToPublish.Add("enabled", "True");
+            authSettingsToPublish.Add("issuer", "https://sts.windows.net/" + tenant + "/");
+            authSettingsToPublish.Add("runtimeVersion", "1.0.0");
+            authSettingsToPublish.Add("tokenStoreEnabled", "True");
+            authSettingsToPublish.Add("unauthenticatedClientAction", "1"); // Corresponds to AllowAnonymous
```

Comment:
You could use the dictionary initialization syntax, it makes it a bit more readable, i.e:

```csharp
var authSettingsToPublish = new Dictionary<string, string>
{
    { "allowedAudiences", serializedArray },
    { "isAadAutoProvisioned", "True" },
    ...
}
```